src/oci/oci_runtime.rs

```diff
@@ -0,0 +1,734 @@
+    /// Set environment variables
+    Env(String),
+    /// Allocate a pseudo-TTY
+    Ttl,
+    /// UID (format: <uid>[:<gid>])
+    User(String),
+    /// Additional gids
+    AdditionalGids(String),
+    /// Path to the process.json
+    Process(PathBuf),
+    /// Detach from the container's process
+    Detach,
+    /// Specify the file to write the process id to
+    PidFile(PathBuf),
+    /// Set the asm process label for the process commonly used with selinux
+    ProcessLabel(String),
+    /// Set the apparmor profile for the process
+    Apparmor(String),
+    /// Set the no new privileges value for the process
+    NoNewPrivs,
+    /// Add a capability to the bounding set for the process
+    Cap(String),
+    /// Pass N additional file descriptors to the container (stdio + $LISTEN_FDS + N in total) (default: 0)
+    PreserveFds(u32),
+}
+
+impl fmt::Display for ExecArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::oci_runtime::ExecArgs::*;
+        write!(f, "--")?;
+
+        match self {
+            ConsoleSocket(path) => write_kv(f, self, path.display()),
+            Cwd(path) => write_kv(f, self, path.display()),
+            Env(env) => write_kv(f, self, env),
+            User(usr) => write_kv(f, self, usr),
+            AdditionalGids(gids) => write_kv(f, self, gids),
+            Process(path) => write_kv(f, self, path.display()),
+            PidFile(path) => write_kv(f, self, path.display()),
+            ProcessLabel(label) => write_kv(f, self, label),
+            Apparmor(app) => write_kv(f, self, app),
+            Cap(cap) => write_kv(f, self, cap),
+            PreserveFds(n) => write_kv(f, self, n),
+            _ => write!(f, "{}", self.as_ref()),
+        }
+    }
+}
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+/// Available arguments for 'oci_runtime kill'.
+pub enum KillArgs {
+    All,
+}
+
+impl fmt::Display for KillArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "--")?;
+        write!(f, "{}", self.as_ref())
+    }
+}
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+/// Available arguments for 'oci_runtime list'.
+pub enum ListArgs {
+    /// One of: table or json (default: "table")
+    Format(String),
+    /// Display only container IDs
+    Quiet,
+}
+
+impl fmt::Display for ListArgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use crate::oci::oci_runtime::ListArgs::*;
+        write!(f, "--")?;
+
+        match self {
+            Format(val) => write_kv(f, self, val),
+            _ => write!(f, "{}", self.as_ref()),
+        }
+    }
+}
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+/// Available arguments for 'oci_runtime ps'.
+pub enum PsArgs {
+    ///one of: table or json (default: "table")
+    Format(String),
```

Comment:
If it's a `one of`, then an `enum` would be probably a better fit than a `String`

Reply:
you are correct. I'll look over the others and see if I can't fix this in a few other places (I think there are a few) Thanks for the awesome code review!